### DISCUSSION TITLE
Multiple assets emit different content to the same filename

### DISCUSSION THREAD

**Discussion by Jakobud** [webpack/webpack#12989] [Q&A] [answered]
Trying to do a really simple SCSS > CSS configuration. Bare minimum configuration here. I have 2 entry files. I want 2 output files. Why is this causing problems?

```
[webpack-cli] Error: Conflict: Multiple chunks emit assets to the same filename file1.css (chunks file1 and file1)
```

```js
'use strict';

const path = require('path');
const MiniCssExtractPlugin = require('mini-css-extract-plugin');

module.exports = {
  mode: 'development',
  output: {
    path: path.resolve('./dist'),
    filename: '[name].css'
  },
  stats: {
    children: true
  },
  name: 'scss',
  entry: {
    file1: path.resolve('./test1.scss'),
    file2: path.resolve('./test2.scss')
  },
  module: {
    rules: [
      {
        test: /(.)*\.scss$/,
        use: [
          MiniCssExtractPlugin.loader,
          'css-loader',
          'postcss-loader',
          {
            loader: 'sass-loader',
            options: {
              implementation: require('node-sass'),
            }
          }
        ]
      }
    ]
  },
  plugins: [
    new MiniCssExtractPlugin()
  ]
}
```

Full repo here: https://github.com/Jakobud/webpack-testing. Just run `npm install` and `npm run webpack`.

**Comment by Jakobud** [comment, marked as answer]
Ah I figured out the typo. Nevermind.

Why can't you delete discussions on Github?

**Reply by alexander-akait** [reply]
Don't know, let's mark as solved :smile:

**Reply by Jakobud** [reply]
Actually I spoke too soon. Still getting the error.

**Reply by alexander-akait** [reply]
Do you use macos?

**Reply by Jakobud** [reply]
I don't think so. Is that a webpack thing?

**Reply by Jakobud** [reply]
With my output as `filename: '[name].css'` I would expect this to output `file1.css` and `file2.css`. Really unclear why this is not working.

**Reply by alexander-akait** [reply]
> I don't think so. Is that a webpack thing?

It is OS, like windows/linux/macos

**Reply by Jakobud** [reply]
I don't have any sort of macros setup in that case. I'm on OSX and I would not know where to go to setup macros.

**Reply by Jakobud** [reply]
Is the repo working fine for you but not for me?

**Reply by alexander-akait** [reply]
Oh, I see, webpack currently doesn't support CSS endpoints with extracting, side effects from this problem https://github.com/webpack/webpack/issues/11671, workaround 
```
new MiniCssExtractPlugin({
  filename: '[name]-bundled.css'
});
```

**Reply by alexander-akait** [reply]
Updated description: https://github.com/webpack/webpack/issues/11671

**Reply by Jakobud** [reply]
Okay I was wondering about this. So is it possible to keep Webpack from generating a file from the `output` config? Or is it just going to always create an empty JS chunk file for each entry file?

I know that Webpack is primarily designed to be using SCSS/CSS in coordination with JS, but in my case there is no JS bundling happening in this configuration.

**Reply by Jakobud** [reply]
Oh and I see, webpack's trying to natively create `file1.css` and then MiniCssExtractPlugin is trying to also create `file1.css`. Okay so yeah, how do I keep Webpack from emitting a file and just leave it to MiniCssExtractPlugin?

**Reply by alexander-akait** [reply]
> So is it possible to keep Webpack from generating a file from the output config?

Unfortunately, this is not possible now, this is a bug/future enchantment 

> Or is it just going to always create an empty JS chunk file for each entry file?

Yes, always

**Reply by alexander-akait** [reply]
> Oh and I see, webpack's trying to natively create file1.css and then MiniCssExtractPlugin is trying to also create file1.css. Okay so yeah, how do I keep Webpack from emitting a file and just leave it to MiniCssExtractPlugin?

Nothing, we should fix it, let's track it in the issue, I think we will fix it in near future, right now just use other name and remove empty JS files manually (yep, bad DX :disappointed: )

**Reply by Jakobud** [reply]
So, I just added this to my plugins and this works fine.

```
new CleanWebpackPlugin({
      cleanAfterEveryBuildPatterns: [
        path.join(process.cwd(), 'dist', '**', '*.js'),
      ]
    }),
```

I know it's really a workaround to just delete stuff after the build, but is there any downside to doing this? The assumption in my case is that I would NEVER have js files in this output directory so deleting all would be fine.

**Reply by alexander-akait** [reply]
> The assumption in my case is that I would NEVER have js files in this output directory so deleting all would be fine.

In this case, all is fine

**Reply by knightswatch3** [reply]
@alexander-akait I actually have the exact same issue for which I raised a discussion. When can I expect the fix in place for this ?

**Reply by Jakobud** [reply]
Looks like there have been discussions on it since 2018. Hopefully it can get solved soon in Webpack 5.

As a workaround I would think it would be possible to make some sort of MiniCssExtractPlugin Clean type of plugin that can maybe hook into when the assets are emitted and figure out which files should be deleted after the build. I'm far from a Webpack expert but it seems like something like that might be possible with what I glance from the API. I mean it would be better if Webpack knew not to emit them in the first place but at least having something to automatically delete them would be helpful.

**Reply by alexander-akait** [reply]
I prioritize this issue, hope we will fix it in near future
